tests/lookup/tests.py

```diff
@@ -45,6 +45,34 @@ def test_exists(self):
         # There should be none now!
         self.assertFalse(Article.objects.exists())
 
+    def test_exists_lookup(self):
+        from django.db.models.expressions import OuterRef
+        Article.objects.filter(author=self.au2).delete()
+        articles = Article.objects.filter(author=OuterRef('pk'))
+        self.assertEquals(self.au1, Author.objects.filter(article__exists=articles).get())
+
+    def test_exists_with_annotate(self):
+        from django.db.models.expressions import Exists, OuterRef
+        Author.objects.annotate(published_after_august=Exists(
+            Article.objects.filter(pub_date__gte='2005-08-01', author=OuterRef('pk')))
+        ).filter(article__exists=Article.objects.filter(author=OuterRef('pk')))
+
+    def _test_exclude_exists_lookup(self):
+        from django.db.models.expressions import OuterRef
+        expected = (
+            'SELECT "lookup_author"."id", "lookup_author"."name" FROM "lookup_author" WHERE NOT EXISTS '
+            '(SELECT "lookup_article"."id" FROM "lookup_article" '
+            'WHERE "lookup_author"."id" = "lookup_article"."lookup_author"."author_id") '
+            'ORDER BY "lookup_author"."name"'
+        )
+        print(expected)
+        result = str(Author.objects.exclude(article__exists=Article.objects.filter(author=OuterRef('pk'))).query)
```

Comment:
How about just Model.objects.filter(Exists(...))? It seems you have the outerq-innerq connection already manually added.

It might be a good choice to skip the lookup support for the initial patch and go with Exists and SubQuery only. Usually if it is possible to write a patch in two parts, it is better to do so.